pkg/apis/kf/v1alpha1/space_types.go

```diff
@@ -95,6 +95,11 @@ type SpaceSpecExecution struct {
 	// +patchMergeKey=name
 	// +patchStrategy=merge
 	Env []corev1.EnvVar `json:"env,omitempty" patchStrategy:"merge" patchMergeKey:"name"`
+
+	// Domains sets valid domains that can be used for routes in the space.
+	// The first entry in the slice is the default if a domain is not
+	// provided.
+	Domains []string
```

Comment:
I think this should be a struct with a `default:true/false` field instead to prevent order from being significant.

You could potentially have zero or more default domains then for apps. That would also prep this to be moved out into a separate CRD